Sources/DeepLinking/Router.swift

```diff
@@ -9,7 +9,9 @@
 import UIKit
 
 public protocol RouteHandler {
-    func handle(route: URL, parameters: [String : String], queryItems: [URLQueryItem])
+    associatedtype T
+
+    func handle(route: URL, parameters: [String : String], queryItems: [URLQueryItem], completion: ((T) -> Void)?)
```

Comment:
Unfortunately we can't do that on protocols yet 😛 

However that's my intention for concrete implementations, yes 😉 

I now noticed that I didn't add that default parameter on `TestHandler`, but I think it was  because the `TreeRouter`'s `route` essentially already does that implicitly, since it's API has the `completion` parameter defaulting to `nil`. I can add it though 😄